home/src/app/pages/search/searchCtrl.js

```diff
@@ -3,92 +3,54 @@
+        vm.itemPerPage = 10;
 
         $scope.DOMAIN_URL = DOMAIN_URL;
-        $scope.listLocations = [];
-        $scope.latlng = [44.841225,-0.580036];
-        $scope.latlon = {"lat":44.841225,"lon":-0.580036};
-        $scope.listLocations.push($scope.latlon);
-
-        // event click on the map add to listLocations
-        $scope.getpos = function(event){
-            var pos = {
-                lat: event.latLng.lat(),
-                lon: event.latLng.lng()
-            };
-            $scope.latlng = [event.latLng.lat(), event.latLng.lng()];
-            $scope.listLocations.push(pos);
-        };
-
-        $scope.searchByLocation = function() {
-            var pos = {
-                lat: _.last($scope.listLocations).lat,
-                lon: _.last($scope.listLocations).lon
-            };
-            console.log(pos);
-
-            $rootScope.searchString = '';
-            $rootScope.selectedLocation = 'all';
-
-            getSearchByLocationData(pos,100000);
+        $scope.allPosts = {};
+        $scope.idPosts = {};
+        $scope.currentPosts = {};
 
-            $timeout( function(){
-               vm.setPage(1);
-            }, 1000 );
+        initController();
 
-        }
+        $scope.$on('handleBroadcast', function() {
+            $scope.searchNull = 0;
+            vm.dummyItems = [];
+            vm.pager = {};
+            initController();
+        });
 
         function initController() {
-            var key = $location.path('/search').search();
-
-            getSearchData(key.searchString,key.location);
-            // initialize to page 1
-            // 10 seconds delay
-            $timeout( function(){
+            if($rootScope.predicates){
                 vm.setPage(1);
-            }, 1000 );
-            
+            } else {
+                $scope.searchNull = 1;
+            }
+
         }
 
         function setPage(page) {
             if (page < 1 || page > vm.pager.totalPages) {
                 return;
             }
-            // get pager object from service
-            vm.pager = PagerService.GetPager(vm.dummyItems.length, page);
-            // get current page of items
-            vm.items = vm.dummyItems.slice(vm.pager.startIndex, vm.pager.endIndex + 1);
-            console.log(vm.items);
-        }
-
-        ////////////////
-        function getSearchData(key,location) {
-             searchService.searchGoods(key,location).then(function (response) {
-                vm.dummyItems = response.data;  
-             }, function () {
-                console.log('Something wrong');
-             });
-        }
-
-        function getSearchByLocationData(pos,distance) {
-            searchService.searchGoodsByLocation(pos.lat,pos.lon,distance).then(function (response) {
-               vm.dummyItems = response.data;
-               console.log(response);
-
+            searchService.predicateSearch($rootScope.predicates).then(function (response) {
```

Comment:
Does paging work on this search page?, I don't see you using 'page' variable at all ?